pkg/settings/inventory.go

```diff
@@ -14,6 +14,7 @@ const (
 	AuthOptional   = "AUTH_OPTIONAL"
 	Host           = "API_HOST"
 	Port           = "API_PORT"
+	TLSSecretName  = "TLS_SECRET_NAME"
```

Comment:
Few things.  The name of the secret would not be sufficient without the namespace.
Instead of the name of the secret, let's have the configmap/operator set the paths explicitly.  This will eliminate the need for hard coded paths and make it much easier to test and run in development environment.
API_TLS_KEY
API_TLS_CERTIFICATE
API TLS_CA

To be set by the operator.  This will k

Reply:
To be mounted, the secret must be in the same namespace as the pod, so the secret name is sufficient.
And the path is already set by the operator and follows the convention `/var/run/secrets/<secret_name>`.
Then, inside the secret, the file names are always `tls.crt` and `tls.key`.

Reply:
I still think explicit paths are better than integration _by-convention_.  This way the operator can change it's convention without breaking the controller.  This decoupling is also useful for developement/testing.